src/main/java/org/openstreetmap/atlas/utilities/vectortiles/TippecanoeCommands.java

```diff
@@ -39,6 +40,55 @@ private TippecanoeCommands()
         }
     };
 
+    /**
+     * Decompresses a directory of *.geojson.gz files recursively into *.geojson files.
+     *
+     * @param geojsonDirectory
+     *            The directory to decompress.
+     */
+    public static void decompress(final Path geojsonDirectory)
+    {
+        final Time time = Time.now();
+        final String directory = geojsonDirectory.toString();
+        final String cat = String.format(
+                "find '%s' -type f -name '*.geojson.gz' -exec gzip -dr \\{\\} +", directory);
+        final String[] bashCommandArray = new String[] { "bash", "-c", cat };
+        try
+        {
+            RunScript.run(bashCommandArray);
+            logger.info("Decompressed line-delimited GeoJSON in {}", time.elapsedSince());
+        }
+        catch (final CoreException exception)
+        {
+            logger.warn("Not finding any .geojson.gz to decompress. Continuing...", exception);
+        }
+    }
+
+    /**
+     * Concatenates all of the GeoJSON files in a directory into a single GeoJSON file. This find
+     * command is fairly sophisticated. What you see here is that it is finding everything in a
+     * certain directory with the extension of *.geojson but is not EVERYTHING.geojosn. It then
+     * pipes those files to cat which in turn outputs to EVERYTHING.geojson.
+     * 
+     * @param geojsonDirectory
+     *            The directory of GeoJSON files to concatenate.
+     */
+    public static void concatenate(final Path geojsonDirectory)
+    {
+        final Time time = Time.now();
+        logger.info("Concatenating GeoJSON...");
+        final String directory = geojsonDirectory.toString();
+        final String cat = String.format(
+                "find '%s' -type f -name '*.geojson' \\! -path '%s'/%s -exec cat \\{\\} + > '%s/'%s",
```

Comment:
As long as the commands calling this function are never run with super-user privileges (or even as a different user than the one invoking it), this is safe.

I am curious what you have in mind in regards to how the `TippecanoeExporter` and the `LineDelimitedGeoJSONConverter` will generally be invoked by end users? I would recommend for whatever program invokes these two commands  to at the very least sanitize its input for characters like `;`.

Reply:
Note that the method is `public static void concatenate(final Path geojsonDirectory)`. At least we know it's a `Path` getting fed in there.

Reply:
True. But someone could still do something like `-geojsonPath="myPath; echo vulnerable"`. However - I took a closer look and as far as I can tell, every user-provided string has been single-quoted, so I think it's actually all good!

I'm just always extra cautious about these things.